fundingmanager_test.go

```diff
@@ -3090,6 +3091,117 @@ func TestFundingManagerFundAll(t *testing.T) {
 	}
 }
 
+// TestFundingManagerFundMax tests that we can initiate a funding request to
+// use the maximum allowed funds remaining in the wallet.
+func TestFundingManagerFundMax(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		coins  []*lnwallet.Utxo
+		change bool
+		wumbo  bool
+	}{
+		{
+			// We will spend all the funds in the wallet, and
+			// expects no change output.
+			coins: []*lnwallet.Utxo{{
+				AddressType: lnwallet.WitnessPubKey,
+				Value:       MaxBtcFundingAmount + 1,
+				PkScript:    coinPkScript,
+				OutPoint: wire.OutPoint{
+					Hash:  chainhash.Hash{},
+					Index: 0,
+				},
+			}},
+			change: false,
+		},
+		{
+			// We spend less than the funds in the wallet,
+			// so a change output should be created.
+			coins: []*lnwallet.Utxo{{
+				AddressType: lnwallet.WitnessPubKey,
+				Value:       2 * MaxBtcFundingAmount,
+				PkScript:    coinPkScript,
+				OutPoint: wire.OutPoint{
+					Hash:  chainhash.Hash{},
+					Index: 0,
+				},
+			}},
+			change: true,
+		},
+		{
+			// All funds in the wallet should be spent with wumbo
+			// active, and therefore we expect no change output.
+			coins: []*lnwallet.Utxo{{
+				AddressType: lnwallet.WitnessPubKey,
+				Value:       2 * MaxBtcFundingAmount,
+				PkScript:    coinPkScript,
+				OutPoint: wire.OutPoint{
+					Hash:  chainhash.Hash{},
+					Index: 0,
+				},
+			}},
+			wumbo: true,
+		},
+	}
+
+	for _, test := range tests {
+		// We set up our mock wallet to control a list of UTXOs that
+		// sum to more than the max channel size.
+		test := test
+		addFunds := func(fundingCfg *fundingConfig) {
+			wc := fundingCfg.Wallet.WalletController
+			wc.(*mockWalletController).utxos = test.coins
+		}
+		alice, bob := setupFundingManagers(t, addFunds)
```

Comment:
By default TestFundingManager is created with `NoWumboChannels` field inactive, which has the effect that Wumbo is enabled but not advertised as an active feature in the TestNode. Test cases that don't have `wumbo` specified are hence indirectly being tested with Wumbo enabled on the local node but not _advertised_ by the remote node.

Alternatively, `NoWumboChannels` could be actived for test cases that don't use Wumbo. But then it would probably be better to add yet another test case that explicitly ensures that the edge case mentioned above is tested for.

Reply:
As mentioned above, I would be in favor of not allowing wumbo channels to be created using the `fundmax` option (at least for the time being).

Reply:
Mh, personally I would prefer this flag to fund up the maximum possible (similar to how the `sendall` flag works), however up to the soft-limit if it is _active_. I feel that letting this flag suddenly respect a soft-limit that is proactively deactivated by turning on Wumbo is a bit strange. However if the general feeling is that for security reasons this makes sense I will revert the commit to always respect this limit.